languages/go/src/main/kotlin/io/vrap/codegen/languages/go/client/ClientFileProducer.kt

```diff
@@ -0,0 +1,312 @@
+                |    req, err := http.NewRequestWithContext(ctx, method, url, data)
+                |    if err != nil {
+                |        return nil, fmt.Errorf("Creating new request: %w", err)
+                |    }
+                |
+                |    if params != nil {
+                |        req.URL.RawQuery = params.Encode()
+                |    }
+                |
+                |    req.Header.Set("Accept", "application/json; charset=utf-8")
+                |    req.Header.Set("Content-Type", "application/json; charset=utf-8")
+                |    for headerName, headerValue := range headers {
+                |        req.Header.Set(headerName, headerValue)
+                |    }
+                |
+                |    if c.logLevel > 0 {
+                |        logRequest(req)
+                |    }
+                |
+                |    resp, err := c.httpClient.Do(req)
+                |    if err != nil {
+                |        return nil, err
+                |    }
+                |
+                |    if c.logLevel > 0 {
+                |        logResponse(resp)
+                |    }
+                |
+                |    return resp, nil
+                |}
+            """.trimMargin()
+        )
+    }
+
+    fun produceClientApiRoot(type: Api): TemplateFile {
+        return TemplateFile(
+            relativePath = "$basePackageName/client_api_root.go",
+            content = """|
+                |$pyGeneratedComment
+                |package $basePackageName
+                |
+                |<${type.subResources("Client")}>
+                |
+            """.trimMargin().keepIndentation()
+        )
+    }
+
+
+    fun produceClientLoggingFile(): TemplateFile {
+        return TemplateFile(
+            relativePath = "$basePackageName/client_logger.go",
+            content = """
+                |$pyGeneratedComment
+                |package $basePackageName
+                |
+                |import (
+                |	"log"
+                |	"net/http"
+                |	"net/http/httputil"
+                |)
+                |
+                |const logRequestTemplate = `DEBUG:
+                |---[ REQUEST ]--------------------------------------------------------
+                |%s
+                |----------------------------------------------------------------------
+                |`
+                |
+                |const logResponseTemplate = `DEBUG:
+                |---[ RESPONSE ]-------------------------------------------------------
+                |%s
+                |----------------------------------------------------------------------
+                |`
+                |
+                |func logRequest(r *http.Request) {
+                |	body, err := httputil.DumpRequestOut(r, true)
```

Comment:
this is actually bad as when i read it right it will log the whole request including headers which contain sensitive data.

Also I don't know if go supports lambdas I would suggest you a different logger pattern to avoid unnecessary calls to the logger. And maybe even use some logging framework as it makes configuration a lot easier.

Reply:
We use this for debugging purposes. This is espescially helpful when developing the terraform provider so that it is easy to introspect what is happening.

This is really not intended to be used in production-like scenario's